### PR TITLE
Fix/beagle init

### LINKED ISSUE #202: Beagle init replaces beagle-service.ts without asking
Beagle React version: 1.7.0

## Steps To Reproduce

1. Run `yarn beagle`
2. Run `yarn beagle` again
3. The program will ask if we really want to replace "app.tsx"
4. Answer no

## The current behavior
The program will not replace `app.tsx`, as we instructed it. But it will replace `beagle-service.ts`.

## The expected behavior
I expected the program to also ask me if I wanted to replace `beagle-service.ts`. Both files are very important and it's really bad if we replace a file that the user has already configured.

### PR DESCRIPTION
**Pull request by jamessonpenhazup**
**- What I did**
Fixed the beagle init script that replaces beagle-service.ts without asking.
**- How I did it**
Added ask to replaces beagle-service.ts.
**- How to verify it**
Run yarn beagle
Run yarn beagle again
The program will ask if we really want to replace "app.tsx"
The program will ask if we really want to replace "beagle-service.ts"
**- Description for the changelog**
Closes #202

### CODE REVIEW

File: cli/init/index.js [hunk 1]
```diff
@@ -14,37 +14,63 @@
...
+      await access(`${path}/beagle/beagle-service.ts`, constants.F_OK)
+      return true
     } catch (error) {
-      console.error(error)
-      process.exit(1)
+      console.error("beagle/beagle-service.ts did not exist, it was created with Beagle settings.")
+      return false
     }
   }
+
```
Comment (arthurbleilzup) [inline cli/init/index.js:27]:
```suggestion
const appTsx = 'app.tsx'
const beagleServicePath = 'beagle/beagle-service.ts'
```

File: cli/init/index.js [hunk 2]
```diff
@@ -14,37 +14,63 @@
...
   const appTsxExists = async (path) => {
     try {
       await access(`${path}/app.tsx`, constants.F_OK)
-      return true
+      await overwriteAppTsx()
     } catch (error) {
       console.error("app.tsx did not exist, it was created with Beagle settings.")
-      return false
+      await createFile(`${dir}/app.tsx`, appContent)
```
Comment (arthurbleilzup) [inline cli/init/index.js:34]:
```suggestion
      await createFile(`${dir}/${appTsx}`, appContent)
```

File: cli/init/index.js [hunk 3]
```diff
@@ -14,37 +14,63 @@
...
     }
   }
-  const appFileExists = await appTsxExists(dir)
-  if (!appFileExists) {
-    createAppTsx(dir, appContent)
+
+  const overwriteAppTsx = async () => {
+    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', async (answer) => {
+      if (`${answer}` === 'y' || `${answer}` === 'Y') {
+        await createFile(`${dir}/app.tsx`, appContent)
```
Comment (arthurbleilzup) [inline cli/init/index.js:51]:
```suggestion
        await createFile(`${dir}/${appTsx}`, appContent)
```

File: cli/init/index.js [hunk 4]
```diff
@@ -14,37 +14,63 @@
...
+      else {
+        process.exit()
+      }
+    })
+  }
+
+  const beagleServiceFileExists = await beagleServiceExists(dir)
+
+  if (!beagleServiceFileExists) {
+    await createFile(`${dir}/beagle/beagle-service.ts`, beagleContent)
```
Comment (arthurbleilzup) [inline cli/init/index.js:63]:
```suggestion
    await createFile(`${dir}/${beagleServicePath}`, beagleContent)
```

File: cli/init/index.js [hunk 5]
```diff
@@ -14,37 +14,63 @@
...
+    await appTsxExists(dir)
   }
   else {
-    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', (answer) => {
+    rl.question('Do you want to replace "beagle-service.ts" content with the Beagle default configuration (y or n)?', async (answer) => {
       if (`${answer}` === 'y' || `${answer}` === 'Y') {
-        createAppTsx(dir, appContent)
+        await createFile(`${dir}/beagle/beagle-service.ts`, beagleContent)
```
Comment (arthurbleilzup) [inline cli/init/index.js:69]:
```suggestion
        await createFile(`${dir}/${beagleServicePath}`, beagleContent)
```

File: cli/init/index.js [hunk 6]
```diff
@@ -14,37 +14,63 @@
...
 
   await mkdir(`${dir}/beagle/`, { recursive: true });
-  await writeFile(`${dir}/beagle/beagle-service.ts`, beagleContent, fileOptions)
 
-  const createAppTsx = async (path, content) => {
+  const beagleServiceExists = async (path) => {
     try {
-      await writeFile(`${path}/app.tsx`, content)
-      process.exit()
+      await access(`${path}/beagle/beagle-service.ts`, constants.F_OK)
```
Comment (arthurbleilzup) [inline cli/init/index.js:20]:
```suggestion
      await access(`${path}/${beagleServicePath}`, constants.F_OK)
```

File: cli/init/index.js [hunk 7]
```diff
@@ -14,37 +14,63 @@
...
+  const beagleServiceFileExists = await beagleServiceExists(dir)
+
+  if (!beagleServiceFileExists) {
+    await createFile(`${dir}/beagle/beagle-service.ts`, beagleContent)
+    await appTsxExists(dir)
   }
   else {
-    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', (answer) => {
+    rl.question('Do you want to replace "beagle-service.ts" content with the Beagle default configuration (y or n)?', async (answer) => {
       if (`${answer}` === 'y' || `${answer}` === 'Y') {
```
Comment (arthurbleilzup) [inline cli/init/index.js:68]:
```suggestion
      if (new RegExp(`^${answer}$`, 'i').test('y')) {
```

File: cli/init/index.js [hunk 8]
```diff
@@ -14,37 +14,63 @@
...
+      process.exit(1)
     }
   }
-  const appFileExists = await appTsxExists(dir)
-  if (!appFileExists) {
-    createAppTsx(dir, appContent)
+
+  const overwriteAppTsx = async () => {
+    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', async (answer) => {
+      if (`${answer}` === 'y' || `${answer}` === 'Y') {
```
Comment (arthurbleilzup) [inline cli/init/index.js:50]:
```suggestion
      if (new RegExp(`^${answer}$`, 'i').test('y')) {
      if (`${answer}` === 'y' || `${answer}` === 'Y') {
```

**Review by arthurbleilzup — CHANGES_REQUESTED** [review]
I saw some room for improvements on your code, making it simpler and more cleaner, please be free to follow the suggestions and test them out!

File: cli/init/index.js [hunk 9]
```diff
@@ -14,37 +14,63 @@
...
+  const beagleServiceExists = async (path) => {
     try {
-      await writeFile(`${path}/app.tsx`, content)
-      process.exit()
+      await access(`${path}/beagle/beagle-service.ts`, constants.F_OK)
+      return true
     } catch (error) {
-      console.error(error)
-      process.exit(1)
+      console.error("beagle/beagle-service.ts did not exist, it was created with Beagle settings.")
```
Comment (igorgiumellizup) [inline cli/init/index.js:23]:
```suggestion
      console.error("beagle/beagle-service.ts not found! Beagle settings used instead")
```

**Review by igorgiumellizup — COMMENTED** [review]
.

File: cli/init/index.js (hunk 9 above)
Comment (Tiagoperes) [reply cli/init/index.js:23]: Why is this an error log? Isn't expected that this file doesn't exist?

File: cli/init/index.js [hunk 10]
```diff
@@ -1,50 +1,82 @@
...
+
+  const overwriteAppTsx = async () => {
+    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', async (answer) => {
+      if (new RegExp(`^${answer}$`, 'i').test('y')) {
+        if (`${answer}` === 'y' || `${answer}` === 'Y') {
+          await createFile(`${dir}/app.tsx`, appContent)
+          console.log("success! all configuration files were created correctly")
+          process.exit()
+        }
+      }
```
Comment (arthurbleilzup) [inline cli/init/index.js:58]:
```suggestion
      if (new RegExp(`^${answer}$`, 'i').test('y')) {
         await createFile(`${dir}/app.tsx`, appContent)
         console.log("success! all configuration files were created correctly")
         process.exit()
      }
```
We have an unnecessary `if` here

File: cli/init/index.js [hunk 11]
```diff
@@ -1,50 +1,82 @@
...
     output: process.stdout,
   })
 
   await mkdir(`${dir}/beagle/`, { recursive: true });
-  await writeFile(`${dir}/beagle/beagle-service.ts`, beagleContent, fileOptions)
 
-  const createAppTsx = async (path, content) => {
+  const beagleServiceExists = async (path) => {
+    const exists = await existsSync(`${path}/${beagleServicePath}`)
+    return exists
```
Comment (arthurbleilzup) [inline cli/init/index.js:22]:
```suggestion
    return await existsSync(`${path}/${beagleServicePath}`)
```
You can return the result directly without a variable

**Review by arthurbleilzup — CHANGES_REQUESTED** [review]
Just some final details and we are good to go

**Review by arthurbleilzup — APPROVED** [review]

**Review by Tiagoperes — CHANGES_REQUESTED** [review]

**Review by Tiagoperes — APPROVED** [review]

**Pull request merged**